src/ngraph/runtime/backend.hpp

```diff
@@ -117,4 +119,23 @@ class ngraph::runtime::Backend
     // \param op_name is the name of the backend specific op
     // \returns a shared pointer to the op if found, else nullptr
     virtual std::shared_ptr<ngraph::Node> get_backend_op(const std::string& op_name, ...);
+
+    /// \brief Returns memory allocator used by backend for host allocations
+    virtual Allocator* get_host_memory_allocator()
+    {
+        return ngraph::runtime::get_ngraph_allocator();
+    }
+
+    /// \brief Set the host memory allocator to be used by the backend
+    /// \param allocator is pointer to host memory allocator object
+    virtual void set_host_memory_allocator(Allocator* allocator) {}
+    /// \brief Returns memory allocator used by backend for device allocations
+    virtual Allocator* get_device_memory_allocator()
```

Comment:
This should be a pure virtual function. Every device must provide their allocator.